src/libraries/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -4251,6 +4251,28 @@ public void Fill_InvalidStartIndexCount_ThrowsArgumentOutOfRangeException(int ar
             AssertExtensions.Throws<ArgumentOutOfRangeException>("count", () => Array.Fill(new string[arrayLength], "", startIndex, count));
         }
 
+        public class Bar
+        {
+        }
+
+        public class Foo
+        {
+        }
+
+        [Fact]
+        public static void Fill_Casting()
+        {
+            Bar[] barArray = new Bar[2];
+            Array.Fill<object>(barArray, new Bar());
+        }
+
+        [Fact]
+        public static void Fill_Casting_Invalid()
+        {
+            Bar[] barArray = new Bar[2];
```

Comment:
Similar to above, but after the call to `Assert.Throws` this unit test should ensure that the contents of the array remain unchanged.